src/PhpImap/Mailbox.php

```diff
@@ -633,7 +649,7 @@ protected function initMailPart(IncomingMail $mail, $partStructure, $partNum, $m
 			}
 		}
 
-		$isAttachment = $partStructure->ifid || isset($params['filename']) || isset($params['name']);
+		$isAttachment = isset($params['filename']) || isset($params['name']);
```

Comment:
I'm not sure that it's gonna work for all cases. Reason why I've used ` $partStructure->ifid ` because in some cases, for some attachments, filename and name was empty. Remove this.

Reply:
So, anyway, there was a cases when `isset($params['filename']) || isset($params['name'])` was not enough. Please remove this change.

Reply:
What is a workaround for this then? I am facing a similar issue where `textPlain` and `textHtml` is empty and I also concluded the same solution.

Reply:
This change worked for me too. Without this `textHtml` and `textPlain` comes as `null` in mails from exchange servers. Is this PR going to be merged?

Reply:
![screen shot 2018-09-20 at 1 56 33 pm](https://user-images.githubusercontent.com/31030044/45805759-099eda80-bcdd-11e8-82a8-bb1376a44a34.png)
How does ifid concludes that if a mail has an attachment btw?

Reply:
Historically `text/plain` and `text/html` parts don't have `id`, but some of the mail clients create `id` and `ifid` attributes for `text/plain` and `text/html` parts and there is no standard which forbid doing it.

Reply:
It only works without checking for $partStructure->ifid, otherwise the body appears blank for me.